pkg/replica/replica.go

```diff
@@ -167,6 +163,13 @@ func construct(readonly bool, size, sectorSize int64, dir, head string, backingF
 		return nil, err
 	}
 
+	// The backing file path can be changed, need to update it after loading
+	// the meta data.
+	r.info.BackingFile = backingFile
+	if backingFile != nil {
+		r.info.BackingFilePath = backingFile.Path
+	}
```

Comment:
`info.BackingFile` is part of the snapshot file chain and used to help the backing file read inside the replica server. Whereas `info.BackingFile.Path` can [be read in the sync agent server](https://github.com/longhorn/longhorn-engine/blob/1481ab6dd315557d10c7e51081b966b39d694ed0/pkg/backup/main.go#L90) during the backup creation, so that the sync agent server doesn't need to find a way to init the struct `*BackingFile`